packages/stream-caip10-link-handler/src/caip10-link-handler.ts

```diff
@@ -59,23 +59,25 @@ export class Caip10LinkHandler implements StreamHandler<Caip10Link> {
       throw new Error('Caip10Link genesis commit cannot have data')
     }
 
+    const metadata = commit.header
+    if (!(metadata.controllers && metadata.controllers.length === 1)) {
+      throw new Error('Exactly one controller must be specified')
+    }
+    metadata.controller = metadata.controllers[0]
```

Comment:
I don't understand why we want to duplicate data in existing stream types.

Reply:
maintaining consistency between streamtypes without breaking backwards compatibility for existing apps